src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -127,6 +128,18 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return c;
 	}
 
+	/** Eigen3 matrix C = alpha*A + beta*B method */
+	template <typename T>
+	SGMatrix<T> add_impl(const SGMatrix<T>& a, const SGMatrix<T>& b, T alpha, T beta) const
```

Comment:
Again, I think all add and scale methods should be inplace. If a user wants a new matrix where things are added, he can allocate it himeself